### PR TITLE
Add Windows/Linux no-dependency builds

### PR DESCRIPTION
**Pull request by sethrj**
The "build with zero dependencies" use case is helpful for trying to get up and running on a completely new experimental system. Adding a Windows build with error checking improves our ability to catch subtle mistakes, and improves downstream compatibility with SCALE (@hartsw ) and other users.

A follow-on request will fix the errors from no-JSON and MSVC.

### CODE REVIEW

File: src/orange/MatrixUtils.cc [hunk 1]
```diff
@@ -16,6 +16,26 @@ using Mat3 = celeritas::SquareMatrixReal3;
 
 namespace celeritas
 {
+namespace
+{
+//---------------------------------------------------------------------------//
+//! Use fused multiply-add for floating point calculations
+template<class T, std::enable_if_t<std::is_floating_point<T>::value, bool> = true>
+T generic_fma(T a, T b, T y)
```
Comment (sethrj) [inline src/orange/MatrixUtils.cc]: I could probably rename this `fma`, add `CELER_FUNCTION` decorators, and put it in `Algorithms.hh`?

File: src/celeritas/optical/OpticalGenData.hh [hunk 2]
```diff
@@ -42,7 +42,7 @@ struct OpticalGenParamsData
 
     bool cerenkov{false};  //!< Whether Cerenkov is enabled
     bool scintillation{false};  //!< Whether scintillation is enabled
-    real_type capacity{0};  //!< Distribution data buffer capacity
+    size_type capacity{0};  //!< Distribution data buffer capacity
```
Comment (sethrj) [inline src/celeritas/optical/OpticalGenData.hh]: @amandalund I assume this was the intent, and that I'm not missing something?

File: .jenkins [hunk 3]
```diff
@@ -2,7 +2,7 @@ pipeline {
   options {
     disableConcurrentBuilds(abortPrevious: true)
     newContainerPerStage()
-    timeout(time: 2, unit: 'HOURS')
+    timeout(time: 7, unit: 'HOURS')
```
Comment (sethrj) [inline .jenkins:5]: Follow-on to #1186

File: scripts/doc-requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Sphinx >= 4, < 7.3
+Sphinx >= 4, ~= 7.3.0
```
Comment (sethrj) [inline scripts/doc-requirements.txt:1]: Follow-on to #1188

File: src/celeritas/optical/OpticalGenData.hh (hunk 2 above)
Comment (amandalund) [reply src/celeritas/optical/OpticalGenData.hh]: Yes, thanks, that should be `size_type`.

**Comment by sethrj** [comment]
Ping @pcanal

File: .github/workflows/build-ultralite.yml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: build-ultralite-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
+  cancel-in-progress: true
+
+jobs:
+  linux:
+    name: ultralite-linux
```
Comment (pcanal) [inline .github/workflows/build-ultralite.yml]: Should we be more specific as the recipe requires `apt` to run properly?
```suggestion
    name: ultralite-ubuntu
```

File: .github/workflows/build-ultralite.yml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      - name: Cache ccache
+        uses: actions/cache@v4
+        with:
+          path: ${{env.CCACHE_DIR}}
+          key: ccache-ultralite-${{github.run_id}}
+          restore-keys: |
+            ccache-ultralite
+      - name: Zero ccache stats
+        run: |
+          ccache -z
```
Comment (pcanal) [inline .github/workflows/build-ultralite.yml:37]: ~If we reset the cache at each run, how does the `ccache` still help?~

Comment (pcanal) [reply .github/workflows/build-ultralite.yml:37]: Nevermind, this reset only the statistics and keeps around the cached products.

**Comment by pcanal** [comment]
Ideally commit 524fbd4dfcc36e5cb6b73e51f57e8ef10f29f712 (and any related addition) would become its own PR as it is distinct in nature (actual code fix) from the rest of the PR (infrastructure/CI extension).

**Comment by sethrj** [comment]
@pcanal Agreed, I can do that.

File: src/orange/MatrixUtils.cc (hunk 1 above)
Comment (pcanal) [reply src/orange/MatrixUtils.cc]: It does make sense.  Then for consistent we will have to consider replacing the other use of `std::fma` (maybe even in non-templated context with `generic_fma`)

File: src/corecel/sys/ScopedSignalHandler.cc [hunk 7]
```diff
@@ -26,7 +26,9 @@ sig_atomic_t volatile g_celer_signal_bits_ = 0;
 //! Set the bit corresponding to a signal
 extern "C" void celer_set_signal(int signal)
 {
+#ifndef _WIN32
```
Comment (pcanal) [inline src/corecel/sys/ScopedSignalHandler.cc]: We should add a comment why it okay to use in linux but not on windows.  Alternative should use
```
 /EHs
Enables standard C++ stack unwinding. Catches only standard C++ exceptions when you use catch(...) syntax. Unless /EHc is also specified, the compiler assumes that functions declared as extern "C" may throw a C++ exception.
```
If that is the behavior consistent with the linux build?

File: .github/workflows/build-ultralite.yml [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+      CCACHE_MAXSIZE: "20Mi"
+      CMAKE_PRESET: ultralite
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get -q -y update
+          sudo apt-get -q -y install \
+            ccache cmake ninja-build
+      - name: Check out Celeritas
```
Comment (pcanal) [inline .github/workflows/build-ultralite.yml:26]: It feels like from there on, the code is not specific to `ultralite` (except for the cache name), if this is correct then in this PR and/or a subsequent PR we should consider factoring the code.

File: src/orange/MatrixUtils.cc (hunk 1 above)
Comment (sethrj) [reply src/orange/MatrixUtils.cc]: Done in the follow-on PR.

File: .github/workflows/build-ultralite.yml (hunk 8 above)
Comment (sethrj) [reply .github/workflows/build-ultralite.yml:26]: It's effectively the same as the `fast` build yes, but I'm not sure how refactoring could improve much...

File: src/corecel/sys/ScopedSignalHandler.cc (hunk 7 above)
Comment (sethrj) [reply src/corecel/sys/ScopedSignalHandler.cc]: Done (for the follow-on PR)

Comment (pcanal) [reply src/corecel/sys/ScopedSignalHandler.cc]: Thanks.  Additional question in the follow-on PR: https://github.com/celeritas-project/celeritas/pull/1200/files#r1578416785

File: .github/workflows/build-ultralite.yml (hunk 8 above)
Comment (pcanal) [reply .github/workflows/build-ultralite.yml:26]: Fair enough.

**Review by pcanal — APPROVED** [review]
LGTM

**Pull request merged**